eng/Packages.Data.props

```diff
@@ -114,20 +114,34 @@
 
   <!-- Track 2 specific versions -->
   <ItemGroup Condition="'$(IsClientLibrary)' == 'true'">
```

Comment:
All of the extension are already in the general item group above as well. While this override some it doesn't eliminate others. Part my issue with this change is trying to avoid duplicate versions across our different package sets. Perhaps we should figure out a way to ensure we don't have duplicate items across the different package types where there are different conflicting versions. Ideally we shouldn't find a package listed more then once in this file.

Reply:
Let me see who references those and try to clean them up.

Reply:
They might be referenced by the track 1 libraries. At a min we should probably make the lists mutually exclusive (i.e. make the larger top list `'$(IsClientLibrary)' != 'true').

Reply:
Making them exclusive might involve a lot of churn that I don't want to include into this PR. I moved all Extensions references to test project only group.

Reply:
This file is also used as the baseline for ApiCompat.

Reply:
We might need to start using a separate file for ApiCompat.